PubNubMembership/Sources/Membership+PubNub.swift

```diff
@@ -0,0 +1,518 @@
+  ///   - filter: Expression used to filter the results. Only objects whose properties satisfy the given expression are returned. The filter language is defined [here](https://www.pubnub.com/docs/swift/stream-filtering-tutorial#filtering-language-definition).
+  ///   - sort: List of properties to sort response objects
+  ///   - limit: The number of objects to retrieve at a time
+  ///   - page: The paging hash strings used for pagination
+  ///   - requestConfig: Custom configuration overrides for this request
+  ///   - completion: Async `Result` of the method call
+  ///     - **Success**: `Tuple` containing an `Array` of `PubNubMembership`, and the next pagination `PubNubHashedPage` (if one exists)
+  ///     - **Failure**: `Error` describing the failure
+  func fetchMemberships(
+    userId: String?,
+    includeCustom: Bool,
+    includeSpaceFields: Bool,
+    includeSpaceCustomFields: Bool,
+    includeTotalCount: Bool,
+    filter: String?,
+    sort: [PubNub.SpaceMembershipSort],
+    limit: Int?,
+    page: PubNubHashedPage?,
+    requestConfig: PubNub.RequestConfiguration,
+    completion: @escaping ((Result<(memberships: [PubNubMembership], next: PubNubHashedPage?), Error>) -> Void)
+  )
+
+  /// Fetch all `PubNubMembership` linked to a specific `PubNubSpace.id`
+  ///
+  /// - Parameters:
+  ///   - spaceId: Unique identifier for the `PubNubSpace`.
+  ///   - includeCustom: Should the `PubNubMembership.custom` properties be included in the response
+  ///   - includeSpaceFields: Should the `PubNubUser` properties be included in the response
+  ///   - includeSpaceCustomFields: Should the `PubNubUser.custom` properties be included in the response
+  ///   - includeTotalCount: Should the next page include total amount of Space to fetch accessed via `next.totalCount`
+  ///   - filter: Expression used to filter the results. Only objects whose properties satisfy the given expression are returned. The filter language is defined [here](https://www.pubnub.com/docs/swift/stream-filtering-tutorial#filtering-language-definition).
+  ///   - sort: List of properties to sort response objects
+  ///   - limit: The number of objects to retrieve at a time
+  ///   - page: The paging hash strings used for pagination
+  ///   - requestConfig: Custom configuration overrides for this request
+  ///   - completion: Async `Result` of the method call
+  ///     - **Success**: `Tuple` containing an `Array` of `PubNubMembership`, and the next pagination `PubNubHashedPage` (if one exists)
+  ///     - **Failure**: `Error` describing the failure
+  func fetchMemberships(
+    spaceId: String,
+    includeCustom: Bool,
+    includeUserFields: Bool,
+    includeUserCustomFields: Bool,
+    includeTotalCount: Bool,
+    filter: String?,
+    sort: [PubNub.UserMembershipSort],
+    limit: Int?,
+    page: PubNubHashedPage?,
+    requestConfig: PubNub.RequestConfiguration,
```

Comment:
@CraigLn is this parameter optional? If so, shouldn't there be a question mark?

Reply:
It is not optional, but the implementation uses a default value of `.init()` (which is a shortened way of saying `PubNub.RequestConfiguration()`